### DISCUSSION TITLE
Question about simulate the quantum network with a single quantum processor

### DISCUSSION THREAD

**Discussion by poig** [mgg39/qiskit-networks#13] [Ideas]
So firstly we are thinking about simulating in such a way, but how to make a quantum channel with it, if doing swap gate will have a lot of unwanted noise.
![image](https://user-images.githubusercontent.com/43441314/177131774-ff9273ee-29cc-4aeb-beb1-803ebeec6b54.png)

so we should simulate with the same qubit set so we can simulate quantum channel noise with something like `RelaxationNoisePass`, using Control Flow Operations to simulate quantum repeater or QKD. also easier to design our quantum device for simulating physical implementation of the quantum data center and making eve action.
Example:
```
from qiskit.providers.aer import AerSimulator
from qiskit.providers.aer.noise.passes import RelaxationNoisePass

from qiskit import transpile
from qiskit. circuit import Delay, QuantumCircuit
from qiskit.circuit.library.standard_gates import SXGate, CXGate

circ = QuantumCircuit(2)
circ.h(0)
circ.cx(0, 1)
circ.measure_all()
circ = transpile(circ,
                 basis_gates=["rz", "sx", "cx"],
                 scheduling_method="alap",
                 instruction_durations=[
                     ("rz", None, 0),
                     ("sx", None, 160),
                     ("cx", None, 800),
                     ("measure", None, 20000)
                 ])
t1s = t2s = [1e-4, 2e-4]
noise_pass = RelaxationNoisePass(t1s=t1s, t2s=t2s, dt=1e-9, op_types=[CXGate])
noisy_circ = noise_pass(circ)
job = AerSimulator().run(noisy_circ)

circuit:
global phase: π/4
           ┌─────────┐    ┌────┐┌─────────┐     ┌─────────────────┐ ░ ┌─┐   
   q_0: ───┤ Rz(π/2) ├────┤ √X ├┤ Rz(π/2) ├──■──┤ Quantum_channel ├─░─┤M├───
        ┌──┴─────────┴───┐└────┘└─────────┘┌─┴─┐├─────────────────┤ ░ └╥┘┌─┐
   q_1: ┤ Delay(160[dt]) ├─────────────────┤ X ├┤ Quantum_channel ├─░──╫─┤M├
        └────────────────┘                 └───┘└─────────────────┘ ░  ║ └╥┘
meas: 2/═══════════════════════════════════════════════════════════════╩══╩═
                                                                       0  1 
```

**Comment by grageragarces** [comment]
I think having a channel with adjustable noise would be ideal in terms of simulations. That being said in realistic scenarios the noise often makes the systems unusable to begin with. So being able to test topologies against ideal/real/and specific scenarios might greatly benefit researchers or students who are trying to find an efficiency baseline or maximum noise in which their systems are still usable and useful. This also allows for a level of abstraction over hardware, as the user can implement the hardware's parameters directly on top of the system, with it requiring major computational resources
